learn/src/data/woodoku_game_env.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from typing import Tuple, List
+import torch
+
+from woodoku import game
+from woodoku.entity.woodoku_board import WoodokuBoard, BOARD_SIZE
+from woodoku.entity.woodoku_shape import MAX_SHAPE_SIZE, WoodokuShape
+
+
+class Observation:
+    """
+    The class representing the observation.
+
+    The observation space is a 1D tensor that has the following length. One board representation and three shapes to
+    choose from.
+
+    The board representation part will be integer 0 or 1 indicating whether that block is occupied. Each shape will also
+    be 0 or 1 and will reflect the actual shape, and will be totally empty if is not available to choose.
+    """
+
+    dimension: Tuple[int] = (
+        BOARD_SIZE * BOARD_SIZE + MAX_SHAPE_SIZE * MAX_SHAPE_SIZE * 3,
+    )
+    data: torch.Tensor
+
+    def __init__(self) -> None:
+        self.data = torch.zeros(self.dimension)
+
+    def from_game(self, board: WoodokuBoard, shapes: List[WoodokuShape]) -> None:
+        """
+        Update self.data based on the woodoku board and the shapes.
+
+        Args:
+            board: The Woodoku Board.
+            shapes: The shapes to choose.
+        """
+        pass
+
+
+class Action:
```

Comment:
Are random actions or even actions part of the game env? Could just be a function in the Learn package

Reply:
Fixed in [90cc4e8](https://github.com/EdwardHaoranLee/WoodokuLearn/pull/71/commits/90cc4e8f80a58739665d5e00e74f2d249d83b94e)